meson.build

```diff
@@ -30,7 +30,7 @@ project(
 )
 
 if meson.get_compiler('cpp').get_id() == 'gcc' or meson.get_compiler('cpp').get_id() == 'clang'
-  add_project_arguments(['-fvisibility=hidden'], language: 'cpp')
+  add_project_arguments(['-fvisibility=hidden', '-Wno-unused-parameter'], language: 'cpp')
```

Comment:
Since these aren't done upstream in Arrow kind of hard to keep things synced; probably just easiest to ignore until we potentially drop the Arrow testing dependency